### DISCUSSION TITLE
Where does the macos app store downloaded files?

### DISCUSSION THREAD

**Discussion by sottey** [immich-app/immich#14058] [Q&A]
I have immich running on a local server. I have installed the Immich macOS app from the app store. Everything is great but if I select "Download" for an image in the app, it shows the progress bar, presumably downloads the file, but I cannot find it anywhere on my drive.

I have googled, I have looked at the manifest files of the app, I have looked everywhere to figure out where the file is placed when I select "Download" but have made zero progress. 

Does anyone know where the downloaded image is put? If not, is the macos source on github? I can;t find that either? Am I just getting old?

Thanks!

**Comment by danieldietzler** [comment]
<img width="259" alt="Screenshot 2024-11-10 at 17 53 54" src="https://github.com/user-attachments/assets/fa425a41-42d7-488c-80d8-ded96065bd2f">

As you can see that's just the iPad app, so it's possible it's putting it somewhere weird where you can't even access it. We currently don't offer a macOS app.

**Reply by Ender-Wang** [reply]
It would be great to offer a macOS app(just to pull the plug on the Photos app). Since mobile apps are developed in Flutter, offering a desktop app won't require too much work.

**Reply by danieldietzler** [reply]
Oh if that's the case feel free to open a PR :)

**Reply by sottey** [reply]
lol. I LOVE this reply :-)

Having the iPad app is sufficient. As the first commenter said, the mobile apps are in Flutter, so any Apple Silicon device can run them. If it works on macOS, why spend the extra time making yet another release?

**Comment by sottey** [comment]
For those who find this searching for an answer, here you go:

If you are running an iPadOS (or iOS) app on an Apple Silicon macbook, it stores it's files in .../Library/Containers/[appname].

If you run the Immich iPadOS app and select "Download", the files are saved here (replace [USERNAME] with your actual username on the device):

/Users/[USERNAME]/Library/Containers/Immich/Data/Documents

Hope that helps!

**Reply by sottey** [reply]
LOL. I forgot where they were saved, searched and found my own answer. +1 to me for being helpful yet super dumb.
